ehrapy/api/io/_read.py

```diff
@@ -497,14 +498,19 @@ def df_to_anndata(df: pd.DataFrame, columns_obs_only: list[str] | None, index_co
         df = df.set_index(index_column)
     # move columns from the input dataframe to later obs
     dataframes = _move_columns_to_obs(df, columns_obs_only)
+    all_num = False
+    # if data is numerical only, short-circuit AnnData creation to have float dtype instead of object
+    if all(np.issubdtype(column_dtype, np.number) for column_dtype in dataframes.df.dtypes):
```

Comment:
Can't this be shortened to `all_num = all(np.issubdtype(column_dtype, np.number) for column_dtype in dataframes.df.dtypes)`?